File: 프로그래머스/Level 1/덧칠하기.cpp
// https://school.programmers.co.kr/learn/courses/30/lessons/161989
#include <string>
#include <vector>

using namespace std;

int solution(int n, int m, vector<int> section) {
    int answer = 0;
    int maxSection = 0;
    
    for (int & sec: section)
    {
        if (sec < maxSection) continue;
        
        answer++;
        maxSection = sec+m;
    }
    
    return answer;
}

File: 프로그래머스/Level 1/덧칠하기.swift
import Foundation

func solution(_ n:Int, _ m:Int, _ section:[Int]) -> Int {
    var answer: Int = 0, maxIdx: Int = 0
    
    for i in section.indices {
        if section[i] < maxIdx { continue }
        
        answer += 1
        maxIdx = section[i] + m
    }
    return answer
}